Automattic/jetpack
# Update/connect button visual component

This PR creates the ActionButton visual component and uses it in the ConnectButton component of the connection package.

It also creates the Story for visualizing and testing the component using Storybook.

This is part of our efforts to break components into visual components and containers and to have a library visual elements browsable via Storybook

#### Changes proposed in this Pull Request:

* creates the ActionButton visual component and uses it in the ConnectButton component of the connection package. 

#### Jetpack product discussion


p1634153714430900-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






First test if the Connection still works

* Make sure you build the jetpack plugin
* Go to the Jetpack dashboard and make sure you see the Connection screen with the "Set up Jetpack" button
* Click the button and confirm the connection works. Make sure the loader shows up and the button is disabled after clicking
* Make sure there are no glitches in the process, the button should be disabled and displaying the spinner all the time until you get redirected
* sandbox your site to an invalid sandbox and try to connect. Make sure the error message shows below the button.

Test Storybook

* Go to the Story book folder `cd projects/js-packages/storybook`
* `pnpm install`
* pnpm run storybook:dev
* Your browser should open with Storybook
* Go to Playground/Action Button and play around with it. Check if it looks good and the documentation is properly generated
